model.js

```diff
@@ -3,97 +3,135 @@ const { db } = require("./db/connection");
+    if (result.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not found!" });
+    }
+    return result.rows;
+  });
 };
 
+
+
 const fetchArticleComments = (id) => {
-   const sqlQuery = `SELECT * FROM comments
+  const sqlQuery = `SELECT * FROM comments
     WHERE article_id = $1
-    ORDER BY created_at ASC;`
-    return db.query(sqlQuery, [id])
-    .then((result) => {
-        if (result.rows.length === 0){
-          return Promise.reject({status : 404, msg : "Not found!"})
-        } else {
-          return result.rows
-        }
-    })
-}
-
+    ORDER BY created_at ASC;`;
+  return db.query(sqlQuery, [id]).then((result) => {
+    if (result.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not found!" });
+    } else {
+      return result.rows;
+    }
+  });
+};
 
 const fetchSingleArticle = (id) => {
   const sqlQuery = `SELECT * FROM articles
    WHERE article_id = $1`;
-  return db
-    .query(sqlQuery, [id])
-    .then((result) => {
-       
-        if (result.rows.length === 0){
-         return Promise.reject({ status: 404, msg: "Not found!" })
-        }
-        else {
-            return result.rows;
-        }
-    })    
+  return db.query(sqlQuery, [id]).then((result) => {
+    if (result.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not found!" });
+    } else {
+      return result.rows;
+    }
+  });
 };
 
 const sendComment = (id, comment) => {
   const sqlQuery = `
 INSERT INTO comments (article_id, author, body)
 VALUES ($1, (SELECT username FROM users WHERE username = $2), $3)
-RETURNING *;`
+RETURNING *;`;
   return db
-  .query(sqlQuery, [id, comment.username, comment.body])
-  .then((result) => {
-    if (result.rows.length === 0){
-      return Promise.reject({ status: 404, msg: "Not found!" })
-    } else {
-    return result.rows[0]
-    }
-  })
-}
+    .query(sqlQuery, [id, comment.username, comment.body])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not found!" });
+      } else {
+        return result.rows[0];
+      }
+    });
+};
 
 const fetchAllUsers = () => {
-  const sqlQuery = `SELECT * FROM users;`
+  const sqlQuery = `SELECT * FROM users;`;
   return db.query(sqlQuery).then((result) => {
-    return result.rows
-  })
-}
+    return result.rows;
+  });
+};
 
 const changeVotes = (id, votes) => {
-  
-  const sqlQuery = `UPDATE articles SET votes = votes + $2 WHERE article_id = $1 RETURNING*;`
+  const sqlQuery = `UPDATE articles SET votes = votes + $2 WHERE article_id = $1 RETURNING*;`;
 
   return db.query(sqlQuery, [id, votes]).then((result) => {
-    if (result.rows.length === 0){
-      return Promise.reject({status : 404, msg : "Not found!"})
+    if (result.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not found!" });
     } else {
-      return result.rows[0]
+      return result.rows[0];
```

Comment:
Let's focus on drying up your model. At this point we need to consider that we may not need a validTopics variable, as all topics are valid. The big thing that changes is 'WHERE' , so have a look at creating a separate variable for that, which you can add on to your SQL string. This is a tricky one, but you've had to rewrite SQL queries which has made your code open to mistakes. Have a rethink of that please. 😸